### PR TITLE
DLP: Added sample for reidentify free text using FPE and test cases

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [Reidentify Free Text Using FPE](https://cloud.google.com/dlp/docs/samples/dlp-reidentify-text-fpe) and unit tests

### CODE REVIEW

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        string text,
+        string keyName,
+        string wrappedKey,
+        FfxCommonNativeAlphabet alphabet = FfxCommonNativeAlphabet.Numeric,
+        InfoType surrogateType = null)
+    {
+        // Instantiate the client.
+        var dlp = DlpServiceClient.Create();
+
+        // Construct the infoType which will be used as surrogate type and infoType both.
```
Comment (catherinerosesantos) [inline dlp/api/Snippets/ReidentifyFreeTextFpe.cs:36]: Move "both"

which will be used as both surrogate type and infoType

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                {
+                    CryptoKeyName = keyName,
+                    WrappedKey = ByteString.FromBase64(wrappedKey)
+                }
+            },
+            CommonAlphabet = alphabet,
+            SurrogateInfoType = infotype
+        };
+
+        // Construct reidentify config using crypto replace config.
```
Comment (catherinerosesantos) [inline dlp/api/Snippets/ReidentifyFreeTextFpe.cs:54]: CryptoReplaceFfxFpeConfig

**Comment by catherinerosesantos** [comment]
I reviewed the code comments--not the code itself.

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs (hunk 2 above)
Comment (catherinerosesantos) [reply dlp/api/Snippets/ReidentifyFreeTextFpe.cs:54]: ... using the CryptoReplaceFfxFpeConfig transformation method.

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                        {
+                            CryptoReplaceFfxFpeConfig = cryptoReplaceFfxFpeConfig,
+                        },
+                        InfoTypes = { infotype }
+                    }
+                }
+            }
+        };
+
+        // Construct the inspect config using custom info types.
```
[the remark this answers is not in the document]
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/ReidentifyFreeTextFpe.cs]: Done.

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs (hunk 2 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/ReidentifyFreeTextFpe.cs]: Done

File: dlp/api/Snippets/ReidentifyFreeTextFpe.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                {
+                    CryptoKeyName = keyName,
+                    WrappedKey = ByteString.FromBase64(wrappedKey)
+                }
+            },
+            CommonAlphabet = alphabet,
+            SurrogateInfoType = infotype
+        };
+
+        // Construct re-identify config using CryptoReplaceFfxFpeConfig transformation method.
```
Comment (catherinerosesantos) [inline dlp/api/Snippets/ReidentifyFreeTextFpe.cs:54]: Sorry, a few more edits here:

Construct the re-identify config using the CryptoReplaceFfxFpeConfig transformation method.

Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/ReidentifyFreeTextFpe.cs]: Done thanks.

**Review by annadata — APPROVED** [review]

**Review by amanda-tarafa — APPROVED** [review]

**Pull request merged**